common/define-grammar.js

```diff
@@ -67,6 +68,12 @@ module.exports = function defineGrammar(dialect) {
       [$._expression, $.generic_type],
       [$._expression, $.predefined_type],
 
+      [$._conditionable_type, $.type_parameter],
+      [$._parameter_name, $._conditionable_type],
+      [$._conditionable_type, $.generic_type],
+      [$._expression, $._conditionable_type],
+      [$._parameter_name, $._expression, $._conditionable_type],
```

Comment:
Do these conflicts *replace* some of the existing conflicts that previously on `_primary_type`? It would be good to check if we could remove some of those.